lib/mochaReporter.js

```diff
@@ -298,7 +298,6 @@ class ReportportalAgent extends Mocha.reporters.Base {
       this.attributes.delete(this.currentTest.tempId);
       this.descriptions.delete(this.currentTest.tempId);
       this.testCaseIds.delete(this.currentTest.tempId);
-      this.currentTest = null;
```

Comment:
Hi, @NeginMn!
Unfortunately, we can not just delete this line of code and make everything work as expected.
Here is the reason:
Mocha provides the status `pending` for a test without a callback or the tests with `descripe.skip` or `this.skip()`(you can read more about it [here](https://mochajs.org/#pending-tests))

If we remove assignment `this.currentTest = null` this line of code, wouldn't be executed
<img width="772" alt="Screenshot 2023-09-13 at 18 08 34" src="https://github.com/reportportal/agent-js-mocha/assets/84175555/02590949-f75a-4464-8094-3161db44343b">
and in the RP we will see the following 
![Screenshot 2023-09-13 at 18 13 08](https://github.com/reportportal/agent-js-mocha/assets/84175555/23e9131d-a406-4773-b417-909af0a6bfe8)
Tests with `pending` status started twice, but only one of them was finished.

But it is not the main problem.
We expected, that logs, attachments and etc, would be reported incide afterEach hook, not in the test itself.
But we have the following 
![Screenshot 2023-09-13 at 18 18 44](https://github.com/reportportal/agent-js-mocha/assets/84175555/96787ca5-dbeb-4b23-b712-2005d59a4b0c)
It is reported directly inside the test itself.
Here is the afterEach hook itself
<img width="930" alt="Screenshot 2023-09-13 at 18 19 40" src="https://github.com/reportportal/agent-js-mocha/assets/84175555/cde55ba3-7b24-4d8b-952a-7dbe552b812b">

Without `this.currentTest = null`, attachments, logs etc., report on the suit level which also not the best option.
If you have an opportunity and desire you can try to implement attaching meta information to the afterEach hook itself.
Unfortunately, currently, our team doesn't have the capacity to implement this feature. But as soon as we find the capacity, we will try to fix it!